controllers/complianceeventsapi/server.go

```diff
@@ -380,9 +426,108 @@ func parseQueryArgs(queryArgs url.Values) (*queryOptions, error) {
 		}
 	}
 
+	authorizedClusterNames, includeAllManagedClusters,
+		err := getAuthorizedClusterNames(ctx, db, parsed, userChangedConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if includeAllManagedClusters {
+		delete(parsed.Filters, "clusters.cluster_id")
+		delete(parsed.Filters, "clusters.name")
+
+		return parsed, nil
+	}
+
+	if len(authorizedClusterNames) == 0 {
+		log.V(1).Info("The service account does not have GET authorization for all")
+
+		return nil, ErrNotAuthorized
+	}
+
+	delete(parsed.Filters, "clusters.cluster_id")
+
+	parsed.Filters["clusters.name"] = authorizedClusterNames
```

Comment:
After talking it over with QE, I think it'd be better if return unauthorized if any of the cluster filters is for a cluster the user does not have access to rather than filter on the subset of what was requested and what they are authorized for.

Reply:
Should we tell users why? "you don't have auths for a subset of clusters" like this? That could make users confused.